snap/info.go

```diff
@@ -65,3 +71,13 @@ type AppInfo struct {
 	Plugs   map[string]*PlugInfo
 	Slots   map[string]*SlotInfo
 }
+
+// StoreInfo provides specific information for a store hosted snap.
+type StoreInfo struct {
+	LastUpdated     time.Time
+	DownloadSha512  string
+	DownloadSize    int64
+	AnonDownloadURL string
+	DownloadURL     string
```

Comment:
@niemeyer no the issue was really that our code was indeed believing that a protected snap would have the anon one empty, but they are both set instead: PR is here

https://github.com/ubuntu-core/snappy/pull/755

we moved to look whether we have a token instead :/